File: lib/src/model/request.dart
import 'package:flutter_zendesk/flutter_zendesk.dart';
import 'package:flutter_zendesk/src/model/comment.dart';
import 'package:flutter_zendesk/src/model/custom_field.dart';
import 'package:json_annotation/json_annotation.dart';

part 'request.g.dart';

@JsonSerializable(anyMap: true, explicitToJson: true)
class ZdkRequest {
  @JsonKey(name: 'created_at')
  final String createdAt;
  @JsonKey(name: 'comment_count')
  final int commentCount;
  @JsonKey(name: 'updated_at')
  final String updatedAt;
  @JsonKey(name: 'public_updated_at')
  final String publicUpdatedAt;
  @JsonKey()
  final String id;
  @JsonKey(name: 'requester_id')
  final int requesterId;
  final String status;
  @JsonKey(name: 'collaborator_ids')
  final List<int> collaboratorIds;
  final String subject;
  final String description;
  @JsonKey(name: 'last_comment')
  final ZdkComment lastComment;

  @JsonKey(name: 'custom_fields', defaultValue: [])
  final List<CustomField> customFields;

  ZdkRequest(
      {this.createdAt,
      this.commentCount,
      this.updatedAt,
      this.publicUpdatedAt,
      this.id,
      this.requesterId,
      this.status,
      this.collaboratorIds,
      this.subject,
      this.description,
      this.lastComment,
      this.customFields = const []});

  factory ZdkRequest.fromJson(Map json) => _$ZdkRequestFromJson(json);

  Map<String, dynamic> toJson() => _$ZdkRequestToJson(this);
}

File: flutter_zendesk/android/src/main/kotlin/net/amond/flutter_zendesk/Request.kt
package net.amond.flutter_zendesk

import kotlinx.serialization.KSerializer
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import zendesk.support.Comment
import zendesk.support.Request

@Serializable
data class RequestDTO(
    @SerialName("created_at")
    val createdAt: String,
    @SerialName("comment_count")
    val commentCount: Int? = 0,
    @SerialName("updated_at")
    val updatedAt: String,
    @SerialName("public_updated_at")
    val publicUpdatedAt: String,
    val id: String?,
    @SerialName("requester_id")
    val requesterId: Long?,
    val status: String?,
    @SerialName("collaborator_ids")
    val collaboratorIds: List<Long>,
    val subject: String?,
    val description: String?,
    @SerialName("last_comment")
    val lastComment: CommentDTO?
)
val requestSerializer: KSerializer<RequestDTO> = RequestDTO.serializer()

fun Request.toDTO(): RequestDTO {
  return RequestDTO(
      this.createdAt.toString(),
      this.commentCount,
      this.updatedAt.toString(),
      this.publicUpdatedAt.toString(),
      this.id,
      this.requesterId,
      this.status?.toString(),
      this.collaboratorIds,
      this.subject,
      this.description,
      this.lastComment?.toDTO()
  )
}


@Serializable
data class CommentDTO(
    val id: Long?
)
val commentSerializer: KSerializer<CommentDTO> = CommentDTO.serializer()

fun Comment.toDTO(): CommentDTO {
  return CommentDTO(this.id)
}